src/feature/nodelist/node_select.c

```diff
@@ -947,7 +947,6 @@ router_choose_random_node(smartlist_t *excludedsmartlist,
   const int need_uptime = (flags & CRN_NEED_UPTIME) != 0;
   const int need_capacity = (flags & CRN_NEED_CAPACITY) != 0;
   const int need_guard = (flags & CRN_NEED_GUARD) != 0;
-  const int weight_for_exit = (flags & CRN_WEIGHT_AS_EXIT) != 0;
```

Comment:
qq: why was this unused?

Reply:
Because choose_good_exit_server_general() contains an outdated copy of parts of:
* router_choose_random_node()
* router_choose_random_node_helper()
* router_add_running_nodes_to_smartlist()
All interspersed with code that checks exits against pending circuits.

So there are probably some checks that aren't being:
* added to the flags for exits, or
* checked by choose_good_exit_server_general().

It would be great to resolve this technical debt, but it's out of scope for this pull request, and Sponsor 55. Let's open a ticket?

Reply:
Actually, there's a quick refactor that fixes 90% of this issue. It requires creating another function, but I think that's better than duplication of a whole bunch of checks.

Then we can open a ticket for the remaining work.

Reply:
See https://trac.torproject.org/projects/tor/ticket/34200